### PR TITLE
Fixed compatible with API v2

### PR DESCRIPTION
**Pull request by thangnc**
- Follow PR #15 
- Re-enabled required columns: `email`, `fname`, `lname`
- Fixed compatible with API v2: replace interests, addtional merge fields...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,38 +13,33 @@ add e-mail to List in MailChimp.
 
 ## Configuration
 
-- **apikey**: Mailchimp API key (string, required)
-- **list_id**: Mailchimp List id (string, required)
-- **double_optin**: control whether to send an opt-in confirmation email (boolean, default: true)
+- **auth_method**: MailChimp auth method (string, `api_key` or `oauth`, default: `oauth`)
```
Comment (ylzen) [inline README.md]: Default --> api_key

File: README.md [hunk 2]
```diff
@@ -31,13 +34,16 @@ out:
   apikey: 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX-XXX'
   list_id: 'XXXXXXXXXX'
   update_existing: false
+  double_optin: false
+  email_column: e mail
```
Comment (sakama) [inline README.md]: `email`?

File: README.md [hunk 3]
```diff
@@ -31,13 +34,16 @@ out:
...
+  fname_column: first name
```
Comment (sakama) [inline README.md]: this column name is valid?

File: README.md [hunk 4]
```diff
@@ -31,13 +34,16 @@ out:
...
+  lname_column: last name
```
Comment (sakama) [inline README.md]: this column name is valid?

**Review by sakama — COMMENTED** [review]
Basically LGTM. I put minor comments.

File: README.md (hunk 2 above)
Comment (thangnc) [reply README.md]: yes, because it is column name, column header, the alias. then no need to fix `email`

File: README.md (hunk 1 above)
Comment (thangnc) [reply README.md]: Noted & fixed

File: README.md (hunk 2 above)
Comment (sakama) [reply README.md]: > it is column name, column header, the alias

What do you mean?
I means space between "e" and "mail" is valid?

File: README.md (hunk 4 above)
Comment (ylzen) [reply README.md]: @sakama it's the column name defined in MailChimp, it is valid. However, it's confusing to show as 'e mail'. @thangnc previous example in README used 'e-mail', can you keep it that way? Thanks!

Comment (ylzen) [reply README.md]: @thangnc Also, previous README had

> email_column: 'e-mail'
> fname_column: 'first name'
> lname_column: 'lname'

Is '' not required in the config?

Comment (thangnc) [reply README.md]: @ylzen in the config i test, no need to add ''. But i think better i should revert the old config for column's names

Comment (ylzen) [reply README.md]: If single quotes are not required, then no need to revert. Please rename 'e mail' to 'e-mail'. Thanks!

File: README.md [hunk 5]
```diff
@@ -31,13 +34,16 @@ out:
...
   merge_fields:
-  - FNAME
-  - LNAME
-  - WEB
+  - website
```
Comment (ylzen) [inline README.md:42]: Does website need to be in upper case?

File: README.md (hunk 4 above)
Comment (thangnc) [reply README.md]: Noted and will fix

File: README.md (hunk 5 above)
Comment (thangnc) [reply README.md:42]: It should be lower case. The plugin will convert to upper case before pushing data

**Comment by thangnc** [comment]
@ylzen PTAL

**Review by ylzen — APPROVED** [review]

**Pull request merged**